--- a/pages/home_pages/home.py
+++ b/pages/home_pages/home.py
@@ -35,6 +35,7 @@
 from ...format import replace_decimal_separator
 from ...key import TYPE_SINGLESIG
 from ...kboard import kboard
+import board  # Android
 from ...themes import theme
 
 
@@ -42,7 +43,11 @@
     """Home is the main menu page of the app"""
 
     def __init__(self, ctx):
-        shtn_reboot_label = t("Shutdown") if kboard.has_battery else t("Reboot")
+        shtn_reboot_label = (  # Android Custom
+            t("Shutdown")
+            if ctx.power_manager.has_battery() or board.config["type"] == "android"
+            else t("Reboot")
+        )
         super().__init__(
             ctx,
             Menu(
