--- a/pages/mnemonic_loader.py
+++ b/pages/mnemonic_loader.py
@@ -64,6 +64,33 @@
             return MENU_CONTINUE
         return status
 
+    # Custom for Android
+    def load_key_from_clipboard(self):
+        from .encryption_ui import decrypt_kef
+
+        data = self.paste_clipboard()
+
+        if data is None:
+            self.flash_error(t("Failed to load"))
+            return MENU_CONTINUE
+
+        words = []
+        try:
+            data_bytes = decrypt_kef(self.ctx, data)
+            if len(data_bytes) in (16, 32):
+                from embit.bip39 import mnemonic_from_bytes
+
+                words = mnemonic_from_bytes(data_bytes).split()
+        except:
+            self.flash_error(t("Failed to load"))
+            return MENU_CONTINUE
+        
+        if not words:
+            self.flash_error(t("Failed to load"))
+            return MENU_CONTINUE
+
+        return self._load_key_from_words(words)    
+
     def load_key_from_camera(self):
         """Handler for the 'load mnemonic'>'via camera' menu item"""
         submenu = Menu(
