--- a/pages/settings_page.py
+++ b/pages/settings_page.py
@@ -53,6 +53,7 @@
     NUM_SPECIAL_2,
     MENU_CONTINUE,
     MENU_EXIT,
+    MENU_SHUTDOWN,  # Android
     ESC_KEY,
     DEFAULT_PADDING,
 )
@@ -122,6 +123,11 @@
             t("Restore factory settings and reboot?"), self.ctx.display.height() // 2
         ):
             self.ctx.display.clear()
+            # Custom for Android
+            from ..settings import store
+
+            store.restore_defaults()
+            return MENU_SHUTDOWN
             try:
                 # Delete settings from SD
                 with SDHandler() as sd:
@@ -253,13 +259,23 @@
         def handler():
             setting_list = settings_namespace.setting_list()
             namespace_list = settings_namespace.namespace_list()
-            items = [
-                (
-                    settings_namespace.label(ns.namespace.split(".")[-1]),
-                    self.namespace(ns),
-                )
-                for ns in namespace_list
-            ]
+            # Android Custom
+            items = []
+            for ns in namespace_list:
+                # Avoid disabled (comented) settings, Android Only
+                try:
+                    name_space_label = settings_namespace.label(ns.namespace.split(".")[-1])
+                except:
+                    name_space_label = None
+                if name_space_label is not None:
+                    items.extend(
+                        [
+                            (
+                                name_space_label,
+                                self.namespace(ns),
+                            )
+                        ]
+                    )
             items.extend(
                 [
                     (
@@ -283,7 +299,8 @@
 
             # Case for security settings
             if settings_namespace.namespace == "settings.security":
-                items.append((t("Tamper Check Code"), self.enter_modify_tc_code))
+                # Android Custom
+                items.append((t("Tamper Check Code"), None))  # self.enter_modify_tc_code
 
             submenu = Menu(self.ctx, items, back_status=back_status)
             index, status = submenu.run_loop()
@@ -463,9 +480,7 @@
                     t("Change theme and reboot?"), self.ctx.display.height() // 2
                 ):
                     self._settings_exit_check()
-                    self.ctx.display.clear()
-                    self.ctx.power_manager.reboot()
-                    return MENU_EXIT  # In case reboot fails
+                    return MENU_SHUTDOWN  # Android custom
                 # Restore previous theme
                 setting.__set__(settings_namespace, starting_category)
                 theme.update()
