--- a/pages/tools.py
+++ b/pages/tools.py
@@ -46,10 +46,10 @@
                 ctx,
                 [
                     (t("Datum Tool"), self.datum_tool),
-                    (t("Device Tests"), self.device_tests),
+                    (t("Device Tests"), None),  # Android custom
                     # (t("Create QR Code"), self.create_qr),
                     (t("Descriptor Addresses"), self.descriptor_addresses),
-                    (t("Flash Tools"), self.flash_tools),
+                    (t("Flash Tools"), None),  # Android custom
                     (t("Remove Mnemonic"), self.rm_stored_mnemonic),
                 ],
             ),
@@ -85,8 +85,11 @@
             if DatumToolMenu(self.ctx).run() == MENU_EXIT:
                 break
 
-        sys.modules.pop("krux.pages.datum_tool")
-        del sys.modules["krux.pages"].datum_tool
+        try:  # Android custom
+            sys.modules.pop("krux.pages.datum_tool")
+            del sys.modules["krux.pages"].datum_tool
+        except KeyError:
+            pass
         return MENU_CONTINUE
 
     # def create_qr(self):
