--- a/settings.py
+++ b/settings.py
@@ -25,7 +25,10 @@
 # from .sd_card import SDHandler
 
 import ujson as json
+from kivy.storage.jsonstore import JsonStore  # Custom for Android
+
 import os
+import board  # Custom for Android
 
 # Settings storage
 SD_PATH = "sd"
@@ -88,10 +91,14 @@
     def __set__(self, obj, value):
         if self.attr == "location":
             store.update_file_location(value)
-        if value == self.default_value:
+        # Custom for Android
+        if board.config["type"] == "android":
+            try:
+                store.set_value(obj.namespace, self.attr, value) # Will edit json key
+            except:
+                store.set(obj.namespace, self.attr, value)  # Will create json key
+        elif value == self.default_value:
             store.delete(obj.namespace, self.attr)  # do not store defaults
-        else:
-            store.set(obj.namespace, self.attr, value)  # do store custom settings
 
 
 class CategorySetting(Setting):
@@ -300,8 +307,43 @@
                     pass
             self.dirty = False
 
-        return persisted
+
+class AndroidStore:
+    """Acts as a simple JSON file store for settings for Android apps"""
+
+    def __init__(self):
+        self.settings = JsonStore(SETTINGS_FILENAME)
+
+    def get(self, namespace, setting_name, default_value):
+        """Loads a setting under the given namespace, returning the default value if not set"""
+        if not namespace in self.settings.keys():
+            self.set(namespace, setting_name, default_value)
+        if not setting_name in self.settings.get(namespace):
+            self.set_value(namespace, setting_name, default_value)
+        return self.settings.get(namespace)[setting_name]
+    
+    def set(self, namespace, setting_name, setting_value):
+        """Stores a setting value under the given namespace"""
+        print("Setting", namespace, setting_name, setting_value)
+        # print(self.settings[namespace])
+        self.settings[namespace] = {setting_name : setting_value}
+
+    def set_value(self,namespace, setting_name, setting_value):
+        """
+        Stores a setting value under the given namespace,
+        creating the namespace if it doesn't exist
+        """
+        self.settings[namespace][setting_name] = setting_value
+        # This weird line below will save to the file
+        self.settings[namespace] = self.settings[namespace]
+
+    def restore_defaults(self):
+        self.settings.clear()
+
+    def save_settings(self):
+        return True
+
 
 
 # Initialize singleton
-store = Store()
+store = AndroidStore()
